silabs-pti/src/main/java/com/silabs/pti/Main.java

```diff
@@ -311,4 +321,73 @@ private int runCommandSequence(final CommandLine cli) throws IOException {
     c.close();
     return 0;
   }
+
+  /**
+   * Convert arguments in properties file to CLI arguments format. 
+   * <br/><br/>
+   * NB: 
+   * 1) expected format is <code>-properties=path_to_properties_file</code>,
+   * where <code>path_to_properties_file</code> may be surrounded in double 
+   * quotes in case whitespace characters exists in path 
+   * 2) keys in properties file must be the same as CLI arguments, meaning they
+   * start with hyphen (e.g. <code>-ip</code> or <code>-delay</code>) 
+   * 3) in properties file, arguments without value will not have value 
+   * after '=' delimiter (e.g. <code>-discover=</code>).
+   * <br/><br/>
+   * Input Example (properties file content):
+   * <code><br/>
+   *  -zeroTimeThreshold=1000000<br/>
+      -format=text<br/>
+      -ip=1.2.3.4,9.8.7.6<br/>
+      -serial0=<br/>
+      -discreteNodeCapture=<br/>
+      -discover=<br/>
+   * </code>
+   *
+   * Output Example:
+   *  <code>-zeroTimeThreshold=1000000 -format=text -ip=1.2.3.4,9.8.7.6 -serial0 -discreteNodeCapture -discover</code>
+   *  <br/>
+   * @param progArgs array of args provided to silabs-pti.jar; <code>progArgs[0]</code> is
+   * argument <code>-properties=path_to_properties_file</code>
+   * @return array of strings with arguments in CLI format; empty array of strings
+   * if invalid input or error processing properties file; never null
+   */
+  private String[] convertPropsToArgs(String[] progArgs) {
+    if (progArgs.length == 1 && progArgs[0].startsWith(PROPERTIES)) {
+      //remove any whitespace and escape chars surrounding file path
+      
+      String propsFile =  progArgs[0].substring(PROPERTIES.length()).trim();
+      if (propsFile.startsWith("\"")) {
+        propsFile = propsFile.substring(1);
+      }
+      if (propsFile.endsWith("\"")) {
+        propsFile = propsFile.substring(0, propsFile.length()-1);
+      }
+      
+      Properties props = new Properties();
+      try (BufferedReader propsReader = new BufferedReader(new FileReader(new File(propsFile)))) {
+        props.load(propsReader);
+
+        String[] args = new String[props.size()];
+        final int[] index = new int[1];
+        index[0] = 0;
+        props.forEach((k,v) -> {
+          if (v instanceof String && !((String)v).isBlank()) {
+            args[index[0]] = k+"="+v;
+          } else {
+            args[index[0]] = k.toString();
+          }
+          index[0]++;
+        });
+        
+        return args;
+      } catch (FileNotFoundException e) {
```

Comment:
I thought "FileNotFoundException" extends IOException, so you would not need a separate block, especially if your handling is the same?

Also: I'd much rather prefer if the WHOLE METHOD would simply throw IOException.

This here is eating exceptions, rather than just make the method retrow it. If you can't do anything useful here than printout, it's usually best to just let it bubble up, and if someone else cares up the call stack, they can deal with it.